### PR TITLE
feat: add Ministry Platform provider

### PR DESCRIPTION
**Pull request by Cdrew92**
https://kb.ministryplatform.com/kb/develop/oauth-2-0

### CODE REVIEW

File: src/MinistryPlatform/Provider.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            'email' => $user['email'],
+            'zoneinfo' => $user['zoneinfo'],
+            'locale' => $user['locale'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:88]: Can you remove this method ?

File: src/MinistryPlatform/Provider.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        
+        $response = $this->getHttpClient()->get($this->getMPUrl().'/oauth/connect/userinfo', [
+            'headers' => [
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:59]:
```suggestion
            \GuzzleHttp\RequestOptions::HEADERS => [
```

File: src/MinistryPlatform/Provider.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public const IDENTIFIER = 'MINISTRYPLATFORM';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['http://www.thinkministry.com/dataplatform/scopes/all', 'openid', 'offline_access'];
+
+    /**
+     * {@inheritdoc}
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:21]: Please add a proper docblock here,`{@inheritdoc}` is only if there is already a parent method

File: src/MinistryPlatform/Provider.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     */
+
+    protected function getMPUrl()
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:24]: Can you rename this method to `baseUrl` instead?

File: src/MinistryPlatform/Provider.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        $response = $this->getHttpClient()->get($this->getMPUrl().'/oauth/connect/userinfo', [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+                'Accept: application/json', 
+                'Content-type: application/json',
+                'Scope: http://www.thinkministry.com/dataplatform/scopes/all openid'
+            ]
+        ]);
+
+        return json_decode($response->getBody(), true);
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:67]:
```suggestion
        return json_decode((string) $response->getBody(), true);
```

File: src/MinistryPlatform/README.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+Configure the package's listener to listen for `SocialiteWasCalled` events.
+
+Add the event to your `listen[]` array in `app/Providers/EventServiceProvider`. See the [Base Installation Guide](https://socialiteproviders.com/usage/) for detailed instructions.
+
+```php
+protected $listen = [
+    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
+        // ... other providers
+        'SocialiteProviders\\MinistryPlatform\\MinistryPlatformExtendSocialite@handle',
```
Comment (lucasmichot) [inline src/MinistryPlatform/README.md:44]:
```suggestion
        \SocialiteProviders\MinistryPlatform\MinistryPlatformExtendSocialite::class.'@handle',
```

File: src/MinistryPlatform/composer.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            "email": "cjdrew.dev@gmail.com"
+        }
+    ],
+    "support": {
+        "issues": "https://github.com/socialiteproviders/providers/issues",
+        "source": "https://github.com/socialiteproviders/providers",
+        "docs": "https://socialiteproviders.com/ministryplatform"
+    },
+    "require": {
+        "php": "^7.3|^8.0",
```
Comment (lucasmichot) [inline src/MinistryPlatform/composer.json:25]:
```suggestion
        "php": "^8.0",
```

File: src/MinistryPlatform/composer.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        "ext-json": "*",
+        "socialiteproviders/manager": "^4.1"
```
Comment (lucasmichot) [inline src/MinistryPlatform/composer.json:27]:
```suggestion
        "socialiteproviders/manager": "^4.4"
```

File: src/MinistryPlatform/Provider.php [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    /**
+     * {@inheritdoc}
+     */
+
+    protected function getMPUrl()
+    {
+        return $this->getConfig('base_url');
+    }
+
+    // /**
```
Comment (lucasmichot) [inline src/MinistryPlatform/Provider.php:29]: Can you remove `//` ?

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Some changes requested

**Comment by Cdrew92** [comment]
@lucasmichot - Issues are resolved!

**Pull request closed without merging**